### ISSUE TITLE
base_address() only works for the first module

### ISSUE THREAD

**Issue by LastExceed** [negaisa/process-memory-reader#1]
`bas_address` only looks up the first module of a process and returns it if the name matches the request, else `None`. thus, requesting any module that isn't the first can never succeed

the problem is here:
https://github.com/vagola/process-memory-reader/blob/a4118806322ef40f0376fdc7a6c3a501923788eb/src/windows.rs#L80
memory is only allocated for a single `HMODULE`, but we actually need a whole array, which then needs to be searched for the requested module

**Comment by vaIgarashi** [comment]
Indeed, it must be an array. I am currently not interested in supporting this project. However, contributions are welcome
